### ISSUE TITLE
Octane Compatibility

### ISSUE THREAD

**Issue by astronomersiva** [astronomersiva/ember-pickr#127]

**Comment by richard-viney** [comment]
Are you aware of any blockers? I'm using `ColorPicker` in an Octane app and haven't run into any problems so far.

Or is this GitHub issue referring to upgrading this addon to use the new Octane features?

**Comment by astronomersiva** [comment]
@richard-viney, you are correct. While this addon would(and should) work in Octane apps, this issue is to track the conversion of this addon itself to the Octane style of things.

**Comment by davideferre** [comment]
I've an issue related to Octane. I'm using ember-pickr inside another component. When the value assigned to the `@default` attribute changes the component doesn't redraw itself and remain with the previous color.

**Comment by richard-viney** [comment]
@davideferre I wouldn't expect that to be specific to Octane per se, and looking at the code it may well be intended behaviour.

The `@default` arg is used on initial setup of the component and never checked again. See here: https://github.com/astronomersiva/ember-pickr/blob/master/addon/components/color-picker.js#L281.

**Comment by davideferre** [comment]
@richard-viney I've added the comment to this issue because I think this is octane related issue because octane doesn't reload the component if it is already loaded but change only the data. For this reason in my component I'm using https://github.com/emberjs/ember-render-modifiers. More generally, there is no way to update the color if the picker has already been drawn, or at least I have not found it in the documentation.

**Comment by richard-viney** [comment]
Re. the colour change after initial render, yes that may well be the case. You could open a bug/feature issue for that, it would be fairly simple to address and would involve using either `didReceiveAttrs()` or possibly the `{{did-update}}` modifier.

I'm no longer using this addon, but would be happy to review a PR.
